### PR TITLE
Added `TableText` label tag component

### PR DESCRIPTION
**Pull request by dat-boris**
## Summary:
One of our main use cases is to render the conversation.

The data stored is raw JSON conversation, we create the `TableText` component,
which is inherited from the `HyperText` component which renders the
conversation in a similar way to how Khanmigo renders it.

See example config in `src/examples/rich_text_table/config.xml`:

```
<View>
  <Labels name="ner" toName="text">
    <Label value="Person"></Label>
    <Label value="Organization"></Label>
    <Label value="Date"></Label>
  </Labels>
  <TableText name="text" value="$text"></TableText>
</View>
```

There are still some bits to wrap up in future:
- Fix annotation interface
- Render LaTex statement

Issue: https://khanacademy.atlassian.net/browse/DI-533

## Test plan:
- `yarn start` and visit localhost:3000 to see how the component works
image

- Include the `yarn build:module` components FE and ensure it works by building our custom image

### CODE REVIEW

File: src/env/development.js [hunk 1]
```diff
@@ -76,7 +77,9 @@ import { ClassificationMixed } from '../examples/classification_mixed';
  */
 // import { AllTypes } from "../examples/all_types";
 
-const data = Buckets;
+// CHANGEME: Edit the example you want below:
+//const data = RichTextHtml;
+const data = RichTextTable;
```
Comment (dat-boris) [inline src/env/development.js:82]: For reviewer: (dev-only) this is for local webpack-dev-server to renders the component in Hot reload the component.

(And all files in `src/examples/**`)

File: src/tags/object/RichText/view.js [hunk 2]
```diff
@@ -14,7 +15,7 @@ import { observe } from 'mobx';
 const DBLCLICK_TIMEOUT = 450; // ms
 const DBLCLICK_RANGE = 5; // px
 
-class RichTextPieceView extends Component {
+export class RichTextPieceView extends Component {
```
Comment (dat-boris) [inline src/tags/object/RichText/view.js]: See documentation on top of table.js for explanation of these changes.

File: src/tags/visual/Collapse.js [hunk 3]
```diff
@@ -55,6 +55,7 @@ const PanelModel = types.model({
     'audio',
     'image',
     'hypertext',
+    'tabletext',
```
Comment (dat-boris) [inline src/tags/visual/Collapse.js:58]: These are required to allow tabletext to be one of the child tag of the `<View/>` or collapsable `<Panel/>

File: src/tags/object/RichText/view.js [hunk 4]
```diff
@@ -363,7 +364,15 @@ class RichTextPieceView extends Component {
         .join(newLineReplacement);
     }
 
-    if (item.inline) {
+    // Map value to component if valueToComponent is provided
+    if (valueToComponent) {
+      const renderedComp = valueToComponent(val);
+
+      // Render this to string since this is injected as html below
+      val = ReactDOMServer.renderToString(renderedComp);
```
Comment (dat-boris) [inline src/tags/object/RichText/view.js:376]: TODO: I'm going to document more information on safety of this component.

File: src/examples/rich_text_table/START.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+cd backend
+pip install -r requirements.txt
+```
+
+# Start
+
+Named entity recognition for a piece of html markup
+
+```bash
+python server.py -c config.json -l ../examples/html_document/config.xml -i ../examples/html_document/tasks.json -o output
```
Comment (wwells) [inline src/examples/rich_text_table/START.md]: **Question**:   What does this do?   I couldn't find a server.py?

File: src/tags/object/RichText/table.js [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+    const errMsg = `Couldn't parse JSON: ${e.message}`;
+
+    console.error(errMsg);
+    return <div className={errClass}>{errMsg}</div>;
+  }
+
+  const itemClass = cn('richtext', { elem: 'table-item' });
+
+  const rowElems = conversations.map((conversation, index) => {
+    const question = conversation[0];
```
Comment (wwells) [inline src/tags/object/RichText/table.js:45]: **Question**:   Trying to compare this to the Thread Interactions imported into the BetterMath project - seems like this works?   

```
["Help me solve this", "Of course! Let's start by looking at the problem."]
```

question = "Help me solve this"
answer = "Of course! Let's start by looking at the problem."

Do you anticipate we'll need to add any validation steps to ensure we can render correctly if the RichTextTable is leveraged?

**Review by wwells — APPROVED** [review]
Thank you for driving this!   Reviewed as carefully as I could, but with an eye for speed and acknowledging my own serious gaps in mobx and .js.

File: src/examples/rich_text_table/START.md (hunk 5 above)
Comment (dat-boris) [reply src/examples/rich_text_table/START.md]: Good question! This README is actually from other example templates.  I dont think this apply to this example, so lemme delete this!

File: src/tags/object/RichText/table.js (hunk 6 above)
Comment (dat-boris) [reply src/tags/object/RichText/table.js:45]: We will expect a specific JSON format (e.g. a list of 2-tuple of question and answers) - though the field will be generated from the import pipeline (e.g. in BetterMath case, this is generated from the SQL import pipeline), so it is kinda already validated :-)

**Review by dat-boris — COMMENTED** [review]
Thanks for the review!

**Pull request merged**